selvpc/resource_selvpc_resell_crossregion_subnet_v2.go

```diff
@@ -0,0 +1,211 @@
+						},
+						"cidr": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"network_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"subnet_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"project_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"vlan_id": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"vtep_ip_address": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceResellCrossRegionSubnetV2Create(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	resellV2Client := config.resellV2Client()
+	ctx := context.Background()
+
+	projectID := d.Get("project_id").(string)
+	regionsOpts, err := expandResellV2CrossRegionOpts(d.Get("regions").(*schema.Set))
+	if err != nil {
+		return errParseCrossRegionSubnetV2Regions(err)
+	}
+	opts := crossregionsubnets.CrossRegionSubnetOpts{
+		CrossRegionSubnets: []crossregionsubnets.CrossRegionSubnetOpt{
+			{
+				Quantity: 1,
+				Regions:  regionsOpts,
+				CIDR:     d.Get("cidr").(string),
+			},
+		},
+	}
+
+	log.Print(msgCreate(objectCrossRegionSubnet, opts))
+	crossRegionSubnetsResponse, _, err := crossregionsubnets.Create(ctx, resellV2Client, projectID, opts)
+	if err != nil {
+		return errCreatingObject(objectCrossRegionSubnet, err)
+	}
+	if len(crossRegionSubnetsResponse) != 1 {
+		return errReadFromResponse(objectCrossRegionSubnet)
+	}
+
+	d.SetId(strconv.Itoa(crossRegionSubnetsResponse[0].ID))
+
+	return resourceResellCrossRegionSubnetV2Read(d, meta)
+}
+
+func resourceResellCrossRegionSubnetV2Read(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	resellV2Client := config.resellV2Client()
+	ctx := context.Background()
+
+	log.Print(msgGet(objectCrossRegionSubnet, d.Id()))
+	crossRegionSubnet, response, err := crossregionsubnets.Get(ctx, resellV2Client, d.Id())
+	if err != nil {
+		if response.StatusCode == http.StatusNotFound {
+			d.SetId("")
+			return nil
+		}
+
+		return errGettingObject(objectCrossRegionSubnet, d.Id(), err)
+	}
+
+	d.Set("cidr", crossRegionSubnet.CIDR)
+	d.Set("status", crossRegionSubnet.Status)
+	d.Set("vlan_id", crossRegionSubnet.VLANID)
+
+	associatedServers := serversMapsFromStructs(crossRegionSubnet.Servers)
+	if err := d.Set("servers", associatedServers); err != nil {
+		log.Print(errSettingComplexAttr("servers", err))
+	}
+
+	associatedSubnets := subnetsMapsFromStructs(crossRegionSubnet.Subnets)
+	if err := d.Set("subnets", associatedSubnets); err != nil {
+		log.Print(errSettingComplexAttr("subnets", err))
+	}
+
+	associatedRegions := regionsMapsFromSubnetsStructs(crossRegionSubnet.Subnets)
+	if err := d.Set("regions", associatedRegions); err != nil {
+		log.Print(errSettingComplexAttr("regions", err))
+	}
+
+	associatedProjectID := associatedSubnets[0]["project_id"]
```

Comment:
This will throw panic if we have empty `associatedSubnets` map or if first map doesn't have the `project_id` key.